addons/mass_mailing/wizard/mail_compose_message.py

```diff
@@ -11,6 +11,14 @@ class MailComposeMessage(models.TransientModel):
     campaign_id = fields.Many2one('utm.campaign', string='Mass Mailing Campaign', ondelete='set null')
     mass_mailing_name = fields.Char(string='Mass Mailing Name', help='If set, a mass mailing will be created so that you can track its results in the Email Marketing app.')
     mailing_list_ids = fields.Many2many('mailing.list', string='Mailing List')
+    model_is_thread = fields.Boolean(compute='_compute_model_is_thread')
+
+    @api.depends('model')
+    def _compute_model_is_thread(self):
+        models = self.env['ir.model'].sudo().search([('model', 'in', self.mapped('model'))])
```

Comment:
As you generally don't have much composer to compute at the same time (especially for a field used in form view), I guess we could de-optimize this code and use tools and shared info for ir.model, like
```
for composer in self:
    model = self.env['ir.model']._get(composer.model)
    composer.model_is_thread = model.is_mail_thread
```
Advantages
  * no direct access to ir.model
  * use cached data for ir.model

Con: could make one query / composer, but generally we have one composer in a given recordset

Reply:
:disguised_face: Both are ok for me :)

Reply:
There might also be something to be said about checking whether the thing is in mass mailing mode or not, because when it's not the query is useless (and not currently performed).

It does make the field more specific though, not sure it's worth the small gain.

```
for composer in self:
     if composer.composition_mode == 'mass_mail':
          composer.display_mailing_option = False
          continue
     ...
```

Reply:
Anyway the query is a quick one, accessing ir.model is normally fast (short table, accessed as read). We can keep things simple :)